### PR TITLE
Deal with dependencies

### LINKED ISSUE #541: CSSMinFilter should not remove spaces from inside calc() expression
CSSMinFilter removes the space from inside of `calc()` expressions, making them invalid.

`calc(100% + 20px)` becomes `calc(100%+20px)`, which is invalid. The spaces are required.

https://developer.mozilla.org/en-US/docs/Web/CSS/calc

"Note: The + and - operators must always be surrounded by whitespace"

### LINKED ISSUE #586: Large CSS file Online Compression crashing Django
Today I was planning on switching from Foundation to Semantic UI. Semantic CSS file is huge compared to say bootstrap or foundation and so I tried to compress it via Online Compression, but Django crashed because compression was taking too long. I've circumvented this by using offline compression which still takes around 1-2 mins but it eventually works. I was wondering why this was happening when doing it during online compression. Also, is there a way to have successful online compression for large files without crashing django? I wasn't sure if this was a bug in the django-compressor itself or was because it was trying to compress a large file in the request/response cycle.

My compressor related settings:

	# Django Compressor Settings
	COMPRESS_ENABLED = True

	COMPRESS_OUTPUT_DIR = ''

	COMPRESS_CSS_FILTERS = [
	     'compressor.filters.cssmin.CSSMinFilter',
	     'compressor.filters.css_default.CssAbsoluteFilter',
	]
	COMPRESS_JS_FILTERS = [
	     'compressor.filters.jsmin.JSMinFilter'
	]

	COMPRESS_OFFLINE = True

### LINKED ISSUE #664: deal with the dependencies
i propose to:
* deprecate beatifulsoup v3. v4 works for all the python versions supported by compressor. support for v4 was done in #663.
* deprecate cssmin. it is unmaintained, has bugs (#541, PR for that is #600) and is slow (#655, probably #586).
* introduce https://github.com/sprymix/csscompressor instead, which is a more up-to-date port of the yuicompressor, specifically it has fixed #541
* make rcssmin and rjsmin a dependency, to not have some copied code lingering around. check beforehand whether there have been customizations to them in the compressor repo
* deprecate csstidy. its last release was in 2007.

questions @diox :
* does that above sound sensible to you?
* should the new dependencies (csscompressor, rcssmin, rjsmin) be optional?
* is there anything else you can think of?

### PR DESCRIPTION
**Pull request by karyon**
fixes #664. as discussed, without any deprecation warnings.

closes #541, closes #600, closes #655, and closes #586 as the cssminfilter is now removed.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ runtests:
 	coverage run --branch --source=compressor `which django-admin.py` test --settings=compressor.test_settings compressor
 
 coveragereport:
-	coverage report --omit=compressor/test*,compressor/filters/jsmin/rjsmin*,compressor/filters/cssmin/cssmin*,compressor/utils/stringformat*
```
Comment (diox) [inline Makefile]: yay

**Comment by diox** [comment]
A few flake8 errors are causing the build to fail:
compressor/parser/beautifulsoup.py:16:84: F821 undefined name 'err'
compressor/tests/test_filters.py:6:1: F401 'textwrap' imported but unused
compressor/tests/test_filters.py:10:1: F401 'unittest' imported but unused
compressor/tests/test_filters.py:16:1: F401 'find_command' imported but unused

**Comment by karyon** [comment]
fixed.

**Comment by diox** [comment]
Awesome!

**Pull request merged**